### DISCUSSION TITLE
How to delete isvc pods immediately

### DISCUSSION THREAD

**Discussion by SeibertronSS** [kubeflow/kfserving#2339] [General]
I found that after isvc was deleted, the pods associated with it remained in Terminating state for a long time. How can I delete these pods immediately.

**Comment by yuzisun** [comment]
The termination time is depending on the timeout setting of the service which is by default 5 min because it needs to wait for the request to drain after the service is signaled to terminate. You could tune the timeout value on inference service spec if you think your service needs shorter timeout. You can refer to the timeout field on [control plane API](https://kserve.github.io/website/0.8/reference/api/#serving.kserve.io/v1beta1.ComponentExtensionSpec).

**Comment by SeibertronSS** [comment]
Thank you



| |
***@***.***
|
|
***@***.***
|




---- Replied Message ----
| From | Dan ***@***.***> |
| Date | 07/16/2022 23:56 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [kserve/kserve] How to delete isvc pods immediately (Discussion #2339) |

The termination time is depending on the timeout setting of the service which is by default 5 min because it needs to wait for the request to drain after the service is signaled to terminate. You could tune the timeout value on inference service spec if you think your service needs shorter timeout. You can refer to timeout field on control plane API.
